metadata/src/main/java/org/apache/kafka/image/ConfigurationsImage.java

```diff
@@ -61,6 +61,15 @@ public Properties configProperties(ConfigResource configResource) {
         }
     }
 
+    public Map<String, String> configMap(ConfigResource configResource) {
```

Comment:
It would be good to add JavaDoc here stating that this doesn't handle configuration overrides. Also, how about `configMapForResource` as a name?